components/places/src/error.rs

```diff
@@ -163,3 +163,158 @@ pub enum InvalidMetadataObservation {
     #[error("Observed view time is invalid (too long)")]
     ViewTimeTooLong,
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum PlacesError {
```

Comment:
I was thinking about this, should we abandon the `error_support` approach before we do this? This will allow us to only have exactly one high-level enum and not one `internal` and one `external`. 

The biggest benefit `error_support` **offers** is backtrace support, but that's disabled and it looks like it's been broken for a while. (I tried experimenting with it a few weeks back with no success). There is a PR I'll open today with removing `error_support` for the `push` component, maybe we can see what that looks like and if its worth it to do that here - or maybe this will be one of the cleanups after uniffication, if getting it done this way is easier!

Reply:
(for ref, here's the PR I have for push https://github.com/mozilla/application-services/pull/4650)

Reply:
I really like the changes you did in Push! I think places should move in that direction but I'd probably err on the side of caution and maybe add this as a task near the end-tail of the uniffication. Mainly just to reduce amount of actual changes while the core APIs are being uniffied -- however I don't see why we wouldn't be able to do this relatively quickly once the dust is settled!